presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ReorderJoins.java

```diff
@@ -98,6 +99,10 @@ public ReorderJoins(CostComparator costComparator)
         }
 
         MultiJoinNode multiJoinNode = toMultiJoinNode(joinNode, lookup);
+        if(multiJoinNode.getSources().stream()
+                .anyMatch(source -> lookup.getStats(source, session, symbolAllocator.getTypes()).equals(PlanNodeStatsEstimate.UNKNOWN_STATS))) {
```

Comment:
Unfortunately, this can be insufficient. Stats can be partial. Collecting table-level stats gives you row counts (is this information sufficient to _"educated guess"_-style decision?). Collecting column-level stats gives you ndvs, nulls, min & max. You can collect column-level stats for a selection of columns... Oh, and ideally, you should not assume that having stats for table is sufficient to compute cost for a plan. Maybe there is a join expression that we can't really compute stats/costs for?

Maybe wherever you ` .min(planNodeOrdering)` (e.g. https://github.com/Teradata/presto/pull/575/files#diff-5cdb22a9680530f837d520839d7eaf99R158) you should first check compared plans have known cost and back off if any does not?
(I didn't yet think how to put this into code, maybe your neat&slim stream-based loop won't be so neat&slim anymore :( )